File: packages/shared/poolmaster_shared/events/scoring.py
"""Scoring-related events."""

from datetime import datetime
from uuid import UUID

from poolmaster_shared.events.base import DomainEvent


class StatEvent(DomainEvent):
    """A stat event ingested from a sports data provider."""

    type: str = "stat.received"
    source_service: str = "ingestion-worker"
    event_id: str
    participant_external_id: str
    participant_id: UUID | None = None
    stat_key: str
    stat_value: float
    round: int | None = None
    is_correction: bool = False
    corrects_event_id: str | None = None
    provider_id: str
    ingested_at: datetime


class ScoreUpdatedEvent(DomainEvent):
    """Published when a team's score changes in a contest."""

    type: str = "score.updated"
    source_service: str = "scoring-service"
    contest_id: UUID
    team_id: UUID
    old_score: float
    new_score: float
    rank: int
    rank_changed: bool

File: packages/shared/events/scoring.ts
import type { DomainEvent } from './base';

/**
 * Live-score persistence event. Emitted by the bus boundary
 * `publishLiveScoreUpdate` after a typed `LiveScoreResult` is validated and
 * its category-specific detail rows are persisted (per plans/117 §10.3).
 *
 * This replaces the legacy `stat.received` event whose untyped
 * `ProviderStatEvent` payload bypassed validation. Subscribers read the
 * persisted detail rows by `(sportEventId, category)` rather than the
 * payload itself.
 */
export interface LiveScorePersistedEvent extends DomainEvent {
  type: 'live_score.persisted';
  sourceService: 'ingestion-worker';
  category: 'GOLF' | 'BASKETBALL' | 'F1' | 'NFL' | 'NASCAR' | 'TENNIS' | 'SOCCER';
  providerId: string;
  /** Number of detail rows persisted in this batch. */
  updatesPersisted: number;
  ingestedAt: string;
}

export interface ScoreUpdatedEvent extends DomainEvent {
  type: 'score.updated';
  sourceService: 'scoring-service';
  contestId: string;
  teamId: string;
  oldScore: number;
  newScore: number;
  rank: number;
  rankChanged: boolean;
}
